### PR TITLE
migrate: aws v2 - vpc lattice

### PR DESCRIPTION
**Pull request by james03160927**
## Description

Partially fixing https://github.com/gruntwork-io/cloud-nuke/issues/770

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

**Comment by wakeful** [comment]
hey @james03160927 

can you run 
```shell
 go generate ./...
```
this should update the the `v2_migration_report/output.md` file.

File: aws/resources/vpc_lattice_service.go [hunk 1]
```diff
@@ -31,6 +33,70 @@ func (network *VPCLatticeService) getAll(_ context.Context, configObj config.Con
...
+	return nil
+}
+func (network *VPCLatticeService) waitUntilAllServiceAssociationDeleted(id *string) error {
+	for i := 0; i < 10; i++ {
+		output, err := network.Client.ListServiceNetworkServiceAssociations(network.Context,&vpclattice.ListServiceNetworkServiceAssociationsInput{
+			ServiceIdentifier: id,
+		})
+
+		if err != nil {
+			return err
```
Comment (denis256) [inline aws/resources/vpc_lattice_service.go:87]: should error be wrapped in `errors.WithStackTrace(err)` ?

**Comment by james03160927** [comment]
Hi @denis256, addressed your comments. PTAL.

**Review by denis256 — DISMISSED** [review]

File: aws/resources/vpc_lattice_service.go [hunk 2]
```diff
@@ -31,6 +33,69 @@ func (network *VPCLatticeService) getAll(_ context.Context, configObj config.Con
...
+	if err := network.waitUntilAllServiceAssociationDeleted(id); err != nil {
+		return err
+	}
+	if err := network.nukeService(id); err != nil {
+		return err
+	}
+
+	return nil
+}
+func (network *VPCLatticeService) waitUntilAllServiceAssociationDeleted(id *string) error {
```
Comment (denis256) [inline aws/resources/vpc_lattice_service.go:79]: Formatting still looks wrong

```
go fmt ./...
```

Comment (james03160927) [reply aws/resources/vpc_lattice_service.go:79]: Hmm should be good now? Can you check?

**Review by denis256 — APPROVED** [review]

**Pull request merged**